### PR TITLE
Component and system fiddling

### PR DESCRIPTION
**Pull request by profxj**
Mainly a new method to convert a list of components (e.g. from
igmguesses) into a list of AbsSystem's.

Includes a test
No new docs other than inline

### CODE REVIEW

File: linetools/isgm/abssystem.py [hunk 1]
```diff
@@ -246,25 +249,50 @@ def add_component(self, abscomp, tol=0.2*u.arcsec, chk_sep=True, **kwargs):
...
+            testcoord = bool(self.coord.separation(abscomp.coord) < tol)
         else:
-            test = True
+            testcoord = True
         # Now redshift/velocity
-        zlim_comp = (1+abscomp.zcomp)*abscomp.vlim/c_mks
-        zlim_sys = (1+self.zabs)*self.vlim/c_mks
-        test = test & (zlim_comp[0] >= zlim_sys[0]) & (zlim_comp[1] <= zlim_sys[1])
+        if chk_z:
+            dz_toler = (1+self.zabs)*vtoler/3e5  # Avoid Quantity for speed
```
Comment (ntejos) [inline linetools/isgm/abssystem.py]: `3e5` -> `c_mks`

**Comment by profxj** [comment]
Anything else?

**Comment by ntejos** [comment]
Did not catch anything else... ;)

File: linetools/isgm/abssystem.py (hunk 1 above)
Comment (profxj) [inline linetools/isgm/abssystem.py]: c_mks has units.  That said, I've replaced 3e5 with c_mks.value

**Comment by profxj** [comment]
Merging

**Pull request merged**